# Importação de Bibliotecas

In [1]:
#install
#!pip install -q -U matplotlib numpy pandas plotly scikit-learn scipy seaborn shap statsmodels xgboost

In [2]:
#imports
import pandas as pd
import numpy as np

# Obtenção da tabela

In [3]:
# obter o csv do arquivo com separação ';'
dataset_cap02 = pd.read_csv('15-dataset/dataset.csv', sep=',')

# Análise Exploratória de Dados (EDA)

In [4]:
# verificar o cabeçario
dataset_cap02.head()

,ID_Pedido,Data_Pedido,ID_Cliente,Segmento,Regiao,Pais,Product ID,Categoria,SubCategoria,Total_Vendas,Quantidade,Desconto,Lucro,Prioridade
0,CA-2012-124891,31-07-2012,RH-19495,Consumidor,New York,United States,TEC-AC-10003033,Tecnologia,Accessories,2309.650,7,0.0,762.1845,Critico
1,IN-2013-77878,05-02-2013,JR-16210,Corporativo,New South Wales,Australia,FUR-CH-10003950,Moveis,Chairs,3709.395,9,0.1,-288.7650,Critico
2,IN-2013-71249,17-10-2013,CR-12730,Consumidor,Queensland,Australia,TEC-PH-10004664,Tecnologia,Phones,5175.171,9,0.1,919.9710,Medio
3,ES-2013-1579342,28-01-2013,KM-16375,Home Office,Berlin,Germany,TEC-PH-10004583,Tecnologia,Phones,2892.510,5,0.1,-96.5400,Medio
4,SG-2013-4320,05-11-2013,RH-9495,Consumidor,Dakar,Senegal,TEC-SHA-10000501,Tecnologia,Copiers,2832.960,8,0.0,311.5200,Critico


In [5]:
#verificar o fim ou calda
dataset_cap02.tail()

,ID_Pedido,Data_Pedido,ID_Cliente,Segmento,Regiao,Pais,Product ID,Categoria,SubCategoria,Total_Vendas,Quantidade,Desconto,Lucro,Prioridade
51285,IN-2014-62366,19-06-2014,KE-16420,Corporativo,Hiroshima,Japan,OFF-FA-10000746,Suprimentos,Fasteners,65.100,5,0.0,4.5000,Medio
51286,US-2014-102288,20-06-2014,ZC-21910,Consumidor,Texas,United States,OFF-AP-10002906,Suprimentos,Appliances,0.444,1,0.8,-1.1100,Medio
51287,US-2013-155768,02-12-2013,LB-16795,Home Office,California,United States,OFF-EN-10001219,Suprimentos,Envelopes,22.920,3,0.0,11.2308,Alto
51288,MX-2012-140767,18-02-2012,RB-19795,Home Office,São Paulo,Brazil,OFF-BI-10000806,Suprimentos,Binders,13.440,2,0.0,2.4000,Medio
51289,MX-2012-134460,22-05-2012,MC-18100,Consumidor,Managua,Nicaragua,OFF-PA-10004155,Suprimentos,Paper,61.380,3,0.0,1.8000,Alto


In [6]:
# informações da tabela
dataset_cap02.info()
print(f"\n{5*'-'} Valores NA por coluna {5*'-'}\n")
print(dataset_cap02.isna().sum())
print(f"\n{5*'-'} Valores duplicados {5*'-'}\n")
print(dataset_cap02.duplicated().sum())

<class 'pandas.DataFrame'>
RangeIndex: 51290 entries, 0 to 51289
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   ID_Pedido     51290 non-null  str    
 1   Data_Pedido   51290 non-null  str    
 2   ID_Cliente    51290 non-null  str    
 3   Segmento      51290 non-null  str    
 4   Regiao        51290 non-null  str    
 5   Pais          51290 non-null  str    
 6   Product ID    51290 non-null  str    
 7   Categoria     51290 non-null  str    
 8   SubCategoria  51290 non-null  str    
 9   Total_Vendas  51290 non-null  float64
 10  Quantidade    51290 non-null  int64  
 11  Desconto      51290 non-null  float64
 12  Lucro         51290 non-null  float64
 13  Prioridade    51290 non-null  str    
dtypes: float64(3), int64(1), str(10)
memory usage: 5.5 MB

----- Valores NA por coluna -----

ID_Pedido       0
Data_Pedido     0
ID_Cliente      0
Segmento        0
Regiao          0
Pais            0
Product ID 

Observamos que a tabela tem:

- 51290 entradas
- 14 colunas
  - 10 str
  - 1 int64
  - 3 float64
- Non-null não teve em nenhuma então quer dizer que todos os dados estão completos


Agora observação da .head():

- 'Total_Vendas','Quantidade','Desconto','Lucro', são colunas numericas
- 'Total_Vendas','Desconto','Lucro', são flutuantes, pois tem casas decimais
- 'Data_Pedido' é do tipo data

Não tem nenhum dado faltante > confirmando o Non-null

Temos 9 dados duplicados

# Limpeza

## Correção dos tipos de dados

In [7]:
#clonagem do dataset
dataset_cap02_limpo = dataset_cap02.copy()

In [8]:
dataset_cap02_limpo['Data_Pedido'] = pd.to_datetime(dataset_cap02_limpo['Data_Pedido'], errors= 'coerce', dayfirst=True, format="%d-%m-%Y")

In [9]:
dataset_cap02_limpo.info()

<class 'pandas.DataFrame'>
RangeIndex: 51290 entries, 0 to 51289
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   ID_Pedido     51290 non-null  str           
 1   Data_Pedido   51290 non-null  datetime64[us]
 2   ID_Cliente    51290 non-null  str           
 3   Segmento      51290 non-null  str           
 4   Regiao        51290 non-null  str           
 5   Pais          51290 non-null  str           
 6   Product ID    51290 non-null  str           
 7   Categoria     51290 non-null  str           
 8   SubCategoria  51290 non-null  str           
 9   Total_Vendas  51290 non-null  float64       
 10  Quantidade    51290 non-null  int64         
 11  Desconto      51290 non-null  float64       
 12  Lucro         51290 non-null  float64       
 13  Prioridade    51290 non-null  str           
dtypes: datetime64[us](1), float64(3), int64(1), str(9)
memory usage: 5.5 MB


In [10]:
dataset_cap02_limpo.head()

,ID_Pedido,Data_Pedido,ID_Cliente,Segmento,Regiao,Pais,Product ID,Categoria,SubCategoria,Total_Vendas,Quantidade,Desconto,Lucro,Prioridade
0,CA-2012-124891,2012-07-31,RH-19495,Consumidor,New York,United States,TEC-AC-10003033,Tecnologia,Accessories,2309.650,7,0.0,762.1845,Critico
1,IN-2013-77878,2013-02-05,JR-16210,Corporativo,New South Wales,Australia,FUR-CH-10003950,Moveis,Chairs,3709.395,9,0.1,-288.7650,Critico
2,IN-2013-71249,2013-10-17,CR-12730,Consumidor,Queensland,Australia,TEC-PH-10004664,Tecnologia,Phones,5175.171,9,0.1,919.9710,Medio
3,ES-2013-1579342,2013-01-28,KM-16375,Home Office,Berlin,Germany,TEC-PH-10004583,Tecnologia,Phones,2892.510,5,0.1,-96.5400,Medio
4,SG-2013-4320,2013-11-05,RH-9495,Consumidor,Dakar,Senegal,TEC-SHA-10000501,Tecnologia,Copiers,2832.960,8,0.0,311.5200,Critico


In [11]:
dataset_cap02_limpo.describe()

,Data_Pedido,Total_Vendas,Quantidade,Desconto,Lucro
count,51290,51290.000000,51290.000000,51290.000000,51290.000000
mean,2013-05-11 21:26:49.155780,246.490581,3.476545,0.142908,28.610982
min,2011-01-01 00:00:00,0.444000,1.000000,0.000000,-6599.978000
25%,2012-06-19 00:00:00,30.758625,2.000000,0.000000,0.000000
50%,2013-07-08 00:00:00,85.053000,3.000000,0.000000,9.240000
75%,2014-05-22 00:00:00,251.053200,5.000000,0.200000,36.810000
max,2014-12-31 00:00:00,22638.480000,14.000000,0.850000,8399.976000
std,NaN,487.565361,2.278766,0.212280,174.340972


In [12]:
dataset_cap02_limpo.describe(include=[str])

,ID_Pedido,ID_Cliente,Segmento,Regiao,Pais,Product ID,Categoria,SubCategoria,Prioridade
count,51290,51290,51290,51290,51290,51290,51290,51290,51290
unique,25035,1590,3,1094,147,10292,3,17,4
top,CA-2014-100111,PO-18850,Consumidor,California,United States,OFF-AR-10003651,Suprimentos,Binders,Medio
freq,14,97,26518,2001,9994,35,31273,6152,29433


## Tratando Valores NaN

## Tratando Valores Duplicados

In [13]:
dataset_cap02_limpo.drop_duplicates(inplace=True)

In [14]:
print(dataset_cap02_limpo.duplicated().sum())

0


In [15]:
dataset_cap02_limpo.info()

<class 'pandas.DataFrame'>
Index: 51281 entries, 0 to 51289
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   ID_Pedido     51281 non-null  str           
 1   Data_Pedido   51281 non-null  datetime64[us]
 2   ID_Cliente    51281 non-null  str           
 3   Segmento      51281 non-null  str           
 4   Regiao        51281 non-null  str           
 5   Pais          51281 non-null  str           
 6   Product ID    51281 non-null  str           
 7   Categoria     51281 non-null  str           
 8   SubCategoria  51281 non-null  str           
 9   Total_Vendas  51281 non-null  float64       
 10  Quantidade    51281 non-null  int64         
 11  Desconto      51281 non-null  float64       
 12  Lucro         51281 non-null  float64       
 13  Prioridade    51281 non-null  str           
dtypes: datetime64[us](1), float64(3), int64(1), str(9)
memory usage: 5.9 MB


## Tratamento de Dados Outliers

# Criando um csv Limpo

In [17]:
dataset_cap02_limpo.to_csv('15-dataset/dataset_limpo.csv', index=False,encoding='utf-8')